server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ConstructingObjectParser;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "name": "analytics",
+ *     "resourceLimits": [
+ *          {
+ *              "resourceName": "jvm",
+ *              "value": 0.4
+ *          }
+ *     ]
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
+
+    public static final int MAX_CHARS_ALLOWED_IN_NAME = 50;
+    private final String name;
+    private final List<ResourceLimit> resourceLimits;
+    private final ResourceLimitGroupMode mode;
+
+    private static final List<String> ALLOWED_RESOURCES = List.of("jvm");
```

Comment:
Given the number of entries in `ALLOWED_RESOURCES` it should not matter much and I would say the arraylist will be faster given it stores only maybe 3-4 values in near future.

Since the member is private and is not initialised/modified anywhere in the class we can mark this as final but it is already immutable.